kannst dir das noch angucken?

```
$ g++ -g -Wall -Werror -pg -std=c++11 *.cpp -o erp
filehandler.cpp: In function ‘void getData(std::fstream&, TArticle*)’:
filehandler.cpp:86:27: error: ‘strtok’ was not declared in this scope
    pch = strtok(cText, ";");
                           ^
filehandler.cpp:93:37: error: ‘strcpy’ was not declared in this scope
      strcpy(AlleArtikel[i].name, pch);
                                     ^
main.cpp: In function ‘int main()’:
main.cpp:23:40: error: too few arguments to function ‘void open(std::string, std::fstream&)’
   std::fstream *database = open(db_path);
                                        ^
In file included from main.cpp:12:0:
filehandler.h:21:6: note: declared here
 void open(std::string filepath, std::fstream& file);
      ^
main.cpp:24:19: error: ‘getLine’ was not declared in this scope
   getLine(database);

```

--- a/filehandler.cpp
+++ b/filehandler.cpp
@@ -5,6 +5,7 @@ provides all needed functions for handling files
 */
 
 #include "filehandler.h"
+#include <string.h>
 #include <iostream>
 
 // Check if filepath exists and can be open
@@ -83,17 +84,17 @@ void getData(std::fstream& file, TArticle AlleArtikel[1000])
 		int n;
 		while (!file.eof()){
 			file.getline(cText, sizeof(cText));
-			pch = std::strtok(cText, ";");
+			pch = strtok(cText, ";");
 			n = 1;
 			while (pch != NULL)
 			{
 				if (n == 1)
 					AlleArtikel[i].id = atoi(pch);
 				else if (n == 2)
-					std::strcpy(AlleArtikel[i].name, pch);
+					strcpy(AlleArtikel[i].name, pch);
 				else if (n == 3)
 					AlleArtikel[i].price = atof(pch);
-				pch = std::strtok(NULL, ";");
+				pch = strtok(NULL, ";");
 				n++;
 			}
 			i++;

--- a/filehandler.cpp
+++ b/filehandler.cpp
@@ -83,17 +83,17 @@ void getData(std::fstream& file, TArticle AlleArtikel[1000])
 		int n;
 		while (!file.eof()){
 			file.getline(cText, sizeof(cText));
-			pch = strtok(cText, ";");
+			pch = std::strtok(cText, ";");
 			n = 1;
 			while (pch != NULL)
 			{
 				if (n == 1)
 					AlleArtikel[i].id = atoi(pch);
 				else if (n == 2)
-					strcpy(AlleArtikel[i].name, pch);
+					std::strcpy(AlleArtikel[i].name, pch);
 				else if (n == 3)
 					AlleArtikel[i].price = atof(pch);
-				pch = strtok(NULL, ";");
+				pch = std::strtok(NULL, ";");
 				n++;
 			}
 			i++;

--- a/main.cpp
+++ b/main.cpp
@@ -16,29 +16,31 @@ handles all stuff
 
 
 int main (void)
-{	
+{
+	TArticle Artikel[1000];
 	std::string db_path = "database.csv";
 	bool exit = false;
 	char input;
-  std::fstream *database = open(db_path);
-  getLine(database);
-  while(!exit)
-  {
-    int methodNumber = menue();
-    switch(methodNumber)
-    {
-      case 1: break;
-      case 2: break;
-      case 3: break;
-      case 4: break;
-      default: break;
-    }
-    std::cout << "Do you want to exit the Tool(y)?" << "\n";
-    std::cin >> input;
-    if(input == 'y' || input == 'Y' || input == 'z' || input == 'Z')
-    {
-      exit = true;
-    }
-  }
+	std::fstream database;
+	open(db_path, database);
+	getData(database, Artikel);
+	while(!exit)
+	{
+		int methodNumber = menue();
+		switch(methodNumber)
+		{
+			case 1: break;
+			case 2: break;
+			case 3: break;
+			case 4: break;
+			default: break;
+		}
+		std::cout << "Do you want to exit the Tool(y)?" << "\n";
+		std::cin >> input;
+		if(input == 'y' || input == 'Y' || input == 'z' || input == 'Z')
+		{
+			exit = true;
+		}
+	}
   return 0;
 }

